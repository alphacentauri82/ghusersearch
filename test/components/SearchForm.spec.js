import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {StyleSheetTestUtils} from 'aphrodite';

import SearchForm from 'components/SearchForm';

describe('Component: SearchForm', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render as expected', () => {
    const component = renderer.create(<SearchForm onSubmit={jest.fn()} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when an initialInputValue prop is absent', () => {
    it('should set inputValue to an empty string', () => {
      const wrapper = shallow(<SearchForm onSubmit={jest.fn()} />);
      expect(wrapper.state('inputValue')).toEqual('');
    });
  });

  describe('when a different initialInputValue is set', () => {
    it('should update the state', () => {
      const wrapper = shallow(
        <SearchForm
          initialInputValue={'hello'}
          onSubmit={jest.fn()}
        />
      );
      expect(wrapper.state('inputValue')).toEqual('hello');
      wrapper.setProps({
        initialInputValue: 'foo',
      });
      expect(wrapper.state('inputValue')).toEqual('foo');
    });
  });

  describe('when the form is submitted', () => {
    describe('and the state has an input value', () => {
      it('should pass the value to the onSubmit function', () => {
        const spy = jest.fn();
        const wrapper = shallow(
          <SearchForm
            onSubmit={spy}
          />
        );
        wrapper.setState({inputValue: 'test'});
        wrapper.simulate('submit', {preventDefault: f => f});
        expect(spy.mock.calls[0]).toMatchSnapshot();
      });
    });

    describe('and the state has no inputValue', () => {
      it('should not call the onSubmit function', () => {
        const spy = jest.fn();
        const wrapper = shallow(
          <SearchForm
            onSubmit={spy}
          />
        );
        wrapper.simulate('submit', {preventDefault: f => f});
        expect(spy).not.toHaveBeenCalled();
      });
    });
  });

});
