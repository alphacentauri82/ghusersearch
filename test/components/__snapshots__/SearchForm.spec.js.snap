// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchForm should render as expected 1`] = `
<form
  className="SearchForm_e296pg"
  onSubmit={[Function]}
>
  <label
    className="u-hiddenVisually"
    htmlFor="searchInput"
  >
    Search for a user
  </label>
  <input
    className="SearchForm_input_12zt507"
    id="searchInput"
    onChange={[Function]}
    placeholder="Search for a user, e.g. simonsmith"
    type="text"
    value=""
  />
  <button
    className="SearchForm_btn_13vgzd5"
    type="submit"
  >
    <svg />
    <span
      className="u-hiddenVisually"
    >
      Submit
    </span>
  </button>
</form>
`;

exports[`Component: SearchForm when the form is submitted and the state has an input value should pass the value to the onSubmit function 1`] = `
Array [
  "test",
]
`;
